### PR TITLE
feat: rewrite the query builder search component

### PR DESCRIPTION
**Pull request by vikrantgupta25**
### Summary

The change enables us to make further improvements in the design and performance / bugs of the query builder where clause. Also it includes the UI changes related to the dropdown and the search bar. Currently there is no usage of the new component in the product

#### Related Issues / PR's

partially fixes - https://github.com/SigNoz/signoz/issues/5150 

#### Screenshots

https://github.com/user-attachments/assets/f5524ba0-d4fa-462c-a8ef-48bd5db27494

### CODE REVIEW

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/CustomDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		currentFilterItem,
+		searchValue,
+		tags,
+		exampleQueries,
+		options,
+		onChange,
+	} = props;
+	const userOs = getUserOperatingSystem();
+	return (
+		<div>
```
Comment (YounixM) [inline frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/CustomDropdown.tsx:42]: React Fragment

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.styles.scss [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+		background: var(--bg-ink-300);
+		padding: 11px 16px;
+		cursor: default;
+
+		.icons {
+			width: 16px;
+			height: 16px;
+			flex-shrink: 0;
+			border-radius: 2.286px;
+			border-top: 1.143px solid var(--Ink-200, #23262e);
```
Comment (YounixM) [inline frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.styles.scss:156]: use correct variables

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.tsx [hunk 3]
```diff
@@ -0,0 +1,857 @@
...
+						searchValue={searchValue}
+						exampleQueries={suggestionsData?.payload?.example_queries || []}
+						tags={tags}
+						currentFilterItem={currentFilterItem}
+					/>
+				)}
+			>
+				{dropdownOptions.map((option) => (
+					<Select.Option
+						key={isObject(option.value) ? JSON.stringify(option.value) : option.value}
```
Comment (YounixM) [inline frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.tsx:833]: Move to a variable and use it in the 2 places

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.tsx [hunk 4]
```diff
@@ -0,0 +1,857 @@
...
+					...prev,
+					key: parsedValue as BaseAutocompleteData,
+					op: '',
+					value: '',
+				}));
+				setCurrentState(DropdownState.OPERATOR);
+				setSearchValue((parsedValue as BaseAutocompleteData)?.key);
+			}
+
+			if (currentState === DropdownState.OPERATOR) {
```
Comment (YounixM) [inline frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.tsx:300]: use else if

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/CustomDropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+	menu: React.ReactElement;
+	searchValue: string;
+	tags: ITag[];
+	options: Option[];
+	exampleQueries: TagFilter[];
+	onChange: (value: TagFilter) => void;
+	currentFilterItem?: ITag;
+}
+
+export default function CustomDropdown(
```
Comment (YounixM) [inline frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/CustomDropdown.tsx:28]: use a better name for the component.

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.styles.scss [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+			margin-left: 12px;
+			gap: 4px;
+		}
+	}
+
+	.search-bar {
+		width: 100%;
+	}
+
+	.ant-tag {
```
Comment (YounixM) [inline frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.styles.scss:201]: try nesting into specific classnames.

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.tsx (hunk 3 above)
Comment (vikrantgupta25) [reply frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.tsx:833]: done

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.tsx (hunk 4 above)
Comment (vikrantgupta25) [reply frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.tsx:300]: done

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.styles.scss (hunk 6 above)
Comment (vikrantgupta25) [reply frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.styles.scss:201]: done

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.styles.scss (hunk 2 above)
Comment (vikrantgupta25) [reply frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.styles.scss:156]: done

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/CustomDropdown.tsx (hunk 1 above)
Comment (vikrantgupta25) [reply frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/CustomDropdown.tsx:42]: done

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/CustomDropdown.tsx (hunk 5 above)
Comment (vikrantgupta25) [reply frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/CustomDropdown.tsx:28]: done

**Review by YounixM — APPROVED** [review]

**Pull request merged**